tiledb/CMakeLists.txt

```diff
@@ -566,6 +566,10 @@ if (TILEDB_AZURE)
             Azure::azure-storage-blobs
             Azure::azure-core
             LibXml2::LibXml2)
+    if(MSVC)
+      # WebServices needed by Azure storage on windows
+      target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE WebServices)
```

Comment:
What is this library?

Reply:
I found I needed this via conda's failed build here: https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=33175&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213

```
   Creating library tiledb.lib and object tiledb.exp
NMAKE : fatal error U1077: 'D:\bld\tiledb_1688760475623\_build_env\Library\bin\cmake.exe' : return code '0xffffffff'
shuffle.c.obj : error LNK2019: unresolved external symbol __imp_pthread_once referenced in function blosc_internal_bitshuffle
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsCreateXmlBuffer referenced in function "public: __cdecl Azure::Storage::_internal::XmlWriter::XmlWriter(void)" (??0XmlWriter@_internal@Storage@Azure@@QEAA@XZ)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsCreateReader referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlReaderContext,struct std::default_delete<struct Azure::Storage::_internal::XmlReaderContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlReaderContext,0>(void)" (??$make_unique@UXmlReaderContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlReaderContext@_internal@Storage@Azure@@U?$default_delete@UXmlReaderContext@_internal@Storage@Azure@@@std@@@0@XZ)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsSetInput referenced in function "public: __cdecl Azure::Storage::_internal::XmlReader::XmlReader(char const *,unsigned __int64)" (??0XmlReader@_internal@Storage@Azure@@QEAA@PEBD_K@Z)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsFreeReader referenced in function "public: __cdecl std::unique_ptr<struct Azure::Storage::_internal::XmlReaderContext,struct std::default_delete<struct Azure::Storage::_internal::XmlReaderContext> >::~unique_ptr<struct Azure::Storage::_internal::XmlReaderContext,struct std::default_delete<struct Azure::Storage::_internal::XmlReaderContext> >(void)" (??1?$unique_ptr@UXmlReaderContext@_internal@Storage@Azure@@U?$default_delete@UXmlReaderContext@_internal@Storage@Azure@@@std@@@std@@QEAA@XZ)
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2'
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsGetReaderProperty referenced in function "public: __cdecl Azure::Storage::_internal::XmlReader::XmlReader(char const *,unsigned __int64)" (??0XmlReader@_internal@Storage@Azure@@QEAA@PEBD_K@Z)
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsGetReaderNode referenced in function "public: struct Azure::Storage::_internal::XmlNode __cdecl Azure::Storage::_internal::XmlReader::Read(void)" (?Read@XmlReader@_internal@Storage@Azure@@QEAA?AUXmlNode@234@XZ)
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2'
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsReadNode referenced in function "public: __cdecl <lambda_526569cda034d7f2252c2a76e57911fa>::operator()(void)const " (??R<lambda_526569cda034d7f2252c2a76e57911fa>@@QEBA@XZ)
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsCreateWriter referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlWriterContext,struct std::default_delete<struct Azure::Storage::_internal::XmlWriterContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlWriterContext,0>(void)" (??$make_unique@UXmlWriterContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlWriterContext@_internal@Storage@Azure@@U?$default_delete@UXmlWriterContext@_internal@Storage@Azure@@@std@@@0@XZ)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsFreeWriter referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlWriterContext,struct std::default_delete<struct Azure::Storage::_internal::XmlWriterContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlWriterContext,0>(void)" (??$make_unique@UXmlWriterContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlWriterContext@_internal@Storage@Azure@@U?$default_delete@UXmlWriterContext@_internal@Storage@Azure@@@std@@@0@XZ)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsSetOutputToBuffer referenced in function "public: __cdecl Azure::Storage::_internal::XmlWriter::XmlWriter(void)" (??0XmlWriter@_internal@Storage@Azure@@QEAA@XZ)
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2'
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteStartElement referenced in function "public: void __cdecl Azure::Storage::_internal::XmlWriter::Write(struct Azure::Storage::_internal::XmlNode)" (?Write@XmlWriter@_internal@Storage@Azure@@QEAAXUXmlNode@234@@Z)
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteStartAttribute referenced in function "public: void __cdecl Azure::Storage::_internal::XmlWriter::Write(struct Azure::Storage::_internal::XmlNode)" (?Write@XmlWriter@_internal@Storage@Azure@@QEAAXUXmlNode@234@@Z)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteEndAttribute referenced in function "public: void __cdecl Azure::Storage::_internal::XmlWriter::Write(struct Azure::Storage::_internal::XmlNode)" (?Write@XmlWriter@_internal@Storage@Azure@@QEAAXUXmlNode@234@@Z)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteXmlBufferToBytes referenced in function "public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl Azure::Storage::_internal::XmlWriter::GetDocument(void)" (?GetDocument@XmlWriter@_internal@Storage@Azure@@QEAA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ)
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2'
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteCharsUtf8 referenced in function "public: void __cdecl Azure::Storage::_internal::XmlWriter::Write(struct Azure::Storage::_internal::XmlNode)" (?Write@XmlWriter@_internal@Storage@Azure@@QEAAXUXmlNode@234@@Z)
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsWriteEndElement referenced in function "public: void __cdecl Azure::Storage::_internal::XmlWriter::Write(struct Azure::Storage::_internal::XmlNode)" (?Write@XmlWriter@_internal@Storage@Azure@@QEAAXUXmlNode@234@@Z)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsCreateError referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlReaderContext,struct std::default_delete<struct Azure::Storage::_internal::XmlReaderContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlReaderContext,0>(void)" (??$make_unique@UXmlReaderContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlReaderContext@_internal@Storage@Azure@@U?$default_delete@UXmlReaderContext@_internal@Storage@Azure@@@std@@@0@XZ)
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\nmake.exe"' : return code '0x2'
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsFreeError referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlReaderContext,struct std::default_delete<struct Azure::Storage::_internal::XmlReaderContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlReaderContext,0>(void)" (??$make_unique@UXmlReaderContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlReaderContext@_internal@Storage@Azure@@U?$default_delete@UXmlReaderContext@_internal@Storage@Azure@@@std@@@0@XZ)
Stop.
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsCreateHeap referenced in function "class std::unique_ptr<struct Azure::Storage::_internal::XmlWriterContext,struct std::default_delete<struct Azure::Storage::_internal::XmlWriterContext> > __cdecl std::make_unique<struct Azure::Storage::_internal::XmlWriterContext,0>(void)" (??$make_unique@UXmlWriterContext@_internal@Storage@Azure@@$$V$0A@@std@@YA?AV?$unique_ptr@UXmlWriterContext@_internal@Storage@Azure@@U?$default_delete@UXmlWriterContext@_internal@Storage@Azure@@@std@@@0@XZ)
azure-storage-common.lib(xml_wrapper.cpp.obj) : error LNK2019: unresolved external symbol WsFreeHeap referenced in function "public: __cdecl std::unique_ptr<struct Azure::Storage::_internal::XmlWriterContext,struct std::default_delete<struct Azure::Storage::_internal::XmlWriterContext> >::~unique_ptr<struct Azure::Storage::_internal::XmlWriterContext,struct std::default_delete<struct Azure::Storage::_internal::XmlWriterContext> >(void)" (??1?$unique_ptr@UXmlWriterContext@_internal@Storage@Azure@@U?$default_delete@UXmlWriterContext@_internal@Storage@Azure@@@std@@@std@@QEAA@XZ)
tiledb.dll : fatal error LNK1120: 21 unresolved externals
```

Reply:
Ah it's a Windows library (and I was wondering why they don't use libxml on Windows). Thanks!